Xenakios/MixerActions.cpp

```diff
@@ -292,15 +297,32 @@ void DoStoreSelTraxHeights(COMMAND_T*)
 
 void DoRecallSelectedTrackHeights(COMMAND_T*)
 {
-	for (int i=0;i<GetNumTracks();i++)
+	if (NF_IsObeyTrackHeightLockEnabled())
 	{
-		MediaTrack* CurTrack = CSurf_TrackFromID(i+1,false);
-		if (*(int*)GetSetMediaTrackInfo(CurTrack,"I_SELECTED",NULL))
+		for (int i = 0; i < GetNumTracks(); i++)
 		{
-			GUID curGUID=*(GUID*)GetSetMediaTrackInfo(CurTrack,"GUID",NULL);
-			for (int j=0;j<(int)g_vec_trackheighs.size();j++)
-				if (GuidsEqual(&curGUID, &g_vec_trackheighs[j].guid))
-					GetSetMediaTrackInfo(CurTrack,"I_HEIGHTOVERRIDE",&g_vec_trackheighs[j].Heigth);	
+			MediaTrack* CurTrack = CSurf_TrackFromID(i + 1, false);
+			if (*(int*)GetSetMediaTrackInfo(CurTrack, "I_SELECTED", NULL) && !!GetMediaTrackInfo_Value(CurTrack, "B_HEIGHTLOCK"))
```

Comment:
This changes height-locked tracks instead of the opposite.

The two branches of the condition can be merged:

```cpp
void DoRecallSelectedTrackHeights(COMMAND_T*)
{
	for (int i = 0; i < GetNumTracks(); i++)
	{
		MediaTrack* CurTrack = CSurf_TrackFromID(i + 1, false);
		if (!GetMediaTrackInfo_Value(CurTrack, "I_SELECTED") || (NF_IsObeyTrackHeightLockEnabled() && GetMediaTrackInfo_Value(CurTrack, "B_HEIGHTLOCK")))
			continue;

		GUID curGUID = *(GUID*)GetSetMediaTrackInfo(CurTrack, "GUID", NULL);
		for (int j = 0; j < (int)g_vec_trackheighs.size(); j++)
			if (GuidsEqual(&curGUID, &g_vec_trackheighs[j].guid))
				GetSetMediaTrackInfo(CurTrack, "I_HEIGHTOVERRIDE", &g_vec_trackheighs[j].Heigth);
	}
```

Shouldn't `DoStoreSelTraxHeights` be also updated to match?